dotnet/src/Experimental/Agents/Internal/Agent.cs

```diff
@@ -119,10 +119,9 @@ internal sealed class Agent : IAgent
         IKernelBuilder builder = Kernel.CreateBuilder();
 
         this.Kernel =
-            Kernel
-                .CreateBuilder()
-                .AddOpenAIChatCompletion(this._model.Model, this._restContext.ApiKey)
-                .Build();
+            this._restContext.HasVersion ?
```

Comment:
So whether we use OpenAI or AOAI depends on the presence of a version in the rest context?

Shouldn't it be the other way around: the presence of a version in the rest context would depend on whether we can OpenAI or AOAI?

I'm not criticizing; I'm actually wondering!

Reply:
This wasn't a principled design decision.  Its just an outgrowth of each service implementation:

- Open AI doesn't require version parameter in endpoint url.
- Azure does.  

We track endpoint state in the rest context...its not explosed publically and we're basically freezing this code for future removal.